index.js

```diff
@@ -1,8 +1,28 @@
+
+const Sha256 = require("@aws-crypto/sha256-js");
+const SignatureV4 = require("@aws-sdk/signature-v4");
+const HttpRequest = require("@aws-sdk/protocol-http");
 const SnappyJS = require("snappyjs");
-const fetch = require("node-fetch");
 const protobuf = require("protobufjs");
-const btoa = (s) => Buffer.from(s, "binary").toString("base64");
 const prom = require("./prom");
+const axios =  require("axios");
+
+// Host and paths
+const awsManagedPrometheusHostname = 'aps-workspaces.us-east-1.amazonaws.com';
+const pathPrefix = '/workspaces/ws-52968a87-d2a6-41ac-8992-065d813ef1da/api/v1';
```

Comment:
Please take this from an env var

Reply:
don't you want to have it as a parameter instead like you said for the creds?

Reply:
Sounds good